clients/web/src/collections/Collection.js

```diff
@@ -70,6 +107,11 @@ var Collection = Backbone.Collection.extend({
      * result when displaying it to the user.
      */
     pageNum: function () {
+        if (this.filterFunc) {
+            if (this.append) { return 0; }
+            this.pageOffsetStack = this.pageOffsetStack || [];
+            return this.pageOffsetStack.length - 1;
```

Comment:
As a personal preference, something unexpected should raise an exception rather than passing a nonsense response back to the user.  In this case, I'm willing accept `-1` as a valid response if it is documented as such.